docs/src/design/chiplets/memory.md

```diff
@@ -220,61 +222,100 @@ $$
-
 >$$
-(1 - n_0) \cdot (1 - n_1) \cdot s'_0 \cdot (1 - s'_1) = 0 \text{ | degree} = 6
+idx0^2 - idx0 = 0 \text{ | degree} = 2
 $$
 
 >$$
-(n_0 + (1 - n_0) \cdot n_1  + (1 - s'_0)) \cdot s'_1 = 0 \text{ | degree} = 5
+idx1^2 - idx1 = 0 \text{ | degree} = 2
 $$
 
-The first constraint enforces that `s_1` is $1$ when the operation is a read and `ctx` and `addr` are both unchanged. The second constraint enforces that when either the context changed, the address changed, or the operation is a write, then `s_1` is set to $0$.
-
 
-To enforce the values of context ID, address, and clock cycle grow monotonically as described in the previous section, we define the following constraint.
+To enforce the values of context ID, word address, and clock cycle grow monotonically as described in the previous section, we define the following constraint.
 
 >$$
-\left(n_0 \cdot \Delta c + (1 - n_0) \cdot (n_1 \cdot \Delta a + (1 - n_1) \cdot \Delta i) \right) - (2^{16} \cdot d_1' + d_0') = 0 \text{ | degree} = 5
+\left(n_0 \cdot \Delta c + (1 - n_0) \cdot (n_1 \cdot \Delta a + (1 - n_1) \cdot \Delta clk) \right) - (2^{16} \cdot d_1' + d_0') = 0 \text{ | degree} = 5
 $$
 
-Where $\Delta i = i' - i - 1$.
+Where $\Delta clk = clk' - clk - 1$.
 
 In addition to this constraint, we also need to make sure that the values in registers $d_0$ and $d_1$ are less than $2^{16}$, and this can be done with [range checks](../range.md).
 
-Next, we need to make sure that values at a given memory address are always initialized to $0$. This can be done with the following constraint:
+Next, we need to ensure that the `f_scw` column is set to $1$ when the context and word address are the same, and $0$ otherwise.
 
->$$
-s_0 \cdot (1 - s_1) \cdot v_i = 0 \text{ for } i \in \{0, 1, 2, 3\} \text{ | degree} = 3
+$$
+f_{scw}' - (1 - (n_0 + (1-n_0) \cdot n_1)) = 0 \text{ | degree} = 4
 $$
 
-Thus, when the operation is a read and either the context changes or the address changes, values in the $v_i$ columns are guaranteed to be zeros.
+Finally, we need to constrain the `v0, v1, v2, v3` columns. We will define a few variables to help in defining the constraints.
 
-Lastly, we need to make sure that for the same context/address combination, the $v_i$ columns of the current row are equal to the corresponding $v_i$ columns of the next row. This can be done with the following constraints:
+$$
+\begin{align*}
+f_0 &= (1 - idx1) \cdot (1 - idx0) \text{ | degree} = 2\\
+f_1 &= (1 - idx1) \cdot idx0 \text{ | degree} = 2\\
+f_2 &= idx1 \cdot (1 - idx0) \text{ | degree} = 2\\
+f_3 &= idx1 \cdot idx0 \text{ | degree} = 2\\
+\end{align*}
+$$
 
->$$
-s_1 \cdot (v_i' - v_i) = 0 \text{ for } i \in \{0, 1, 2, 3\} \text{ | degree} = 2
+The flag $f_i$ is set to $1$ when $v_i$ is being accessed, and $0$ otherwise. Next, for $0 \leq i < 4$,
+
+$$
+c_i = rw' + (1 - rw') \cdot (1 - ew') \cdot (1 - f_i') \text{ | degree} = 4\\
+$$
```

Comment:
There is a naming collision with regards to $c$.

Reply:
Changed all `c` to `ctx`